docs/src/main/asciidoc/application-data-caching-guide.adoc

```diff
@@ -0,0 +1,474 @@
+
+== Solution
+
+We recommend that you follow the instructions in the next sections and create the application step by step.
+However, you can go right to the completed example.
+
+Clone the Git repository: `git clone {quickstarts-clone-url}`, or download an {quickstarts-archive-url}[archive].
+
+The solution is located in the `application-data-caching` directory.
+
+== Creating the Maven project
+
+First, we need to create a new Quarkus project using Maven with the following command:
+
+[source,shell,subs=attributes+]
+----
+mvn io.quarkus:quarkus-maven-plugin:{quarkus-version}:create \
+    -DprojectGroupId=org.acme \
+    -DprojectArtifactId=application-data-caching \
+    -DclassName="org.acme.caching.WeatherForecastResource" \
+    -Dpath="/weather" \
+    -Dextensions="cache,resteasy-jsonb"
+----
+
+This command generates the Maven project with a REST endpoint and imports the `cache` and `resteasy-jsonb` extensions.
+
+== Creating the REST API
+
+Let's start by creating a service that will simulate an extremely slow call to the external meteorological service. Create
+`src/main/java/org/acme/caching/WeatherForecastService.java` with the following content:
+
+[source,java]
+----
+package org.acme.caching;
+
+import java.time.LocalDate;
+
+import javax.enterprise.context.ApplicationScoped;
+
+@ApplicationScoped
+public class WeatherForecastService {
+
+    public String getDailyForecast(LocalDate date, String city) {
+        try {
+            Thread.sleep(2000L); <1>
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+        }
+        return date.getDayOfWeek() + " will be " + getDailyResult(date.getDayOfMonth() % 4) + " in " + city;
+    }
+
+    private String getDailyResult(int dayOfMonthModuloFour) {
+        switch (dayOfMonthModuloFour) {
+            case 0:
+                return "sunny";
+            case 1:
+                return "cloudy";
+            case 2:
+                return "chilly";
+            case 3:
+                return "rainy";
+            default:
+                throw new IllegalArgumentException();
+        }
+    }
+}
+----
+<1> This is where the slowness comes from.
+
+We also need a class that will contain the response sent to the user when he asks for the next three days weather forecast.
+Create `src/main/java/org/acme/caching/WeatherForecast.java` this way:
+
+[source,java]
+----
+package org.acme.caching;
+
+import java.util.List;
+
+public class WeatherForecast {
+
+    private List<String> dailyForecasts;
+
+    private long executionTimeInMs;
+
+    public WeatherForecast(List<String> dailyForecasts, long executionTimeInMs) {
+        this.dailyForecasts = dailyForecasts;
+        this.executionTimeInMs = executionTimeInMs;
+    }
+
+    public List<String> getDailyForecasts() {
+        return dailyForecasts;
+    }
+
+    public long getExecutionTimeInMs() {
+        return executionTimeInMs;
+    }
+
+    // Setters omitted.
```

Comment:
Either they are useful or not but if they are useful, you need them: the user should be able to exactly reproduce the quick start by running through these steps.

Reply:
Removed.